### ISSUE TITLE
CNN Transfer Baseline

### ISSUE THREAD

**Issue by lukhsaankumar** [McMasterAI-Society/DeepFakeDetector#22]
## Issue Type

- [x] **Model**: ML model bug, training issue, or architecture problem
- [ ] **Data**: Dataset issue, preprocessing bug, or data pipeline problem  
- [ ] **Web**: Frontend bug or UI issue in the Next.js dashboard
- [ ] **API**: Backend API bug or FastAPI endpoint issue
- [ ] **Research**: Research question or experimental feature request
- [ ] **Documentation**: Documentation bug or improvement needed
- [ ] **Bug**: General bug fix needed
- [ ] **Enhancement**: New feature or improvement request

## Description
- Use transfer learning to fine-tune a pretrained CNN for deepfake detection. Select a backbone and adapt it for binary classification (real vs AI-generated). Freeze the backbone initially, train a custom classifier head, and optionally unfreeze later layers to fine-tune.

## Deliverable
- Notebook: `notebooks/cnn_transfer_baseline.ipynb`
- Includes model setup, training loop, validation accuracy/F1/AUROC, and brief notes on results

## Additional Context

**Pretrained Model Options:** 
* EfficientNet-B0 — lightweight and accurate
```python
from torchvision.models import efficientnet_b0, EfficientNet_B0_Weights
model = efficientnet_b0(weights=EfficientNet_B0_Weights.IMAGENET1K_V1)
```

* ResNet-50 — classic, fast baseline
```python
from torchvision.models import resnet50, ResNet50_Weights
model = resnet50(weights=ResNet50_Weights.IMAGENET1K_V2)
```

**Comment by Vihaan-Singhal1** [comment]
Notebook added: notebooks/cnn_transfer_baseline.ipynb ✅
PR is also added
